presto-router/src/test/java/com/facebook/presto/router/TestCustomSchedulerManager.java

```diff
@@ -0,0 +1,233 @@
+        }
+
+        @Override
+        public Optional<URI> getDestination(String user, String query)
+        {
+            ++requestsMade;
+            return Optional.of(serverURIs.get(RANDOM.nextInt(serverURIs.size())));
+        }
+
+        @Override
+        public void setCandidates(List<URI> candidates)
+        {
+            this.candidates = candidates;
+        }
+
+        public int getRequestsMade()
+        {
+            return requestsMade;
+        }
+    }
+
+    public static File getRouterConfigFile(List<TestingPrestoServer> servers)
+            throws IOException
+    {
+        Path path = Files.createTempFile("temp-config", ".json");
+        serverURIs = servers.stream()
+                .map(TestingPrestoServer::getBaseUrl)
+                .collect(Collectors.toList());
+        RouterSpec spec = new RouterSpec(ImmutableList.of(new GroupSpec("all", serverURIs, Optional.empty(), Optional.empty())),
+                ImmutableList.of(new SelectorRuleSpec(Optional.empty(), Optional.empty(), Optional.empty(), "all")),
+                Optional.of(CUSTOM_PLUGIN_SCHEDULER),
+                Optional.empty());
+        JsonCodec<RouterSpec> codec = jsonCodec(RouterSpec.class);
+        Files.write(path, codec.toBytes(spec));
+        return path.toFile();
+    }
+
+    private static TestingPrestoServer createPrestoServer()
+            throws Exception
+    {
+        TestingPrestoServer server = new TestingPrestoServer();
+        server.installPlugin(new TpchPlugin());
+        server.createCatalog("tpch", "tpch");
+        server.refreshNodes();
+
+        return server;
+    }
+
+    public void testQuery()
+            throws Exception
+    {
+        String sql = "SELECT row_number() OVER () n FROM tpch.tiny.orders";
+        try (Connection connection = createConnection(httpServerInfo.getHttpUri());
+                Statement statement = connection.createStatement();
+                ResultSet rs = statement.executeQuery(sql)) {
+            long count = 0;
+            long sum = 0;
+            while (rs.next()) {
+                count++;
+                sum += rs.getLong("n");
+            }
+            assertEquals(count, 15000);
+            assertEquals(sum, (count / 2) * (1 + count));
+        }
+
+        MockScheduler mockScheduler = (MockScheduler) schedulerManager.getScheduler();
+        assertEquals(mockScheduler.getRequestsMade(), 1);
+    }
+
+    private static Connection createConnection(URI uri)
+            throws SQLException
+    {
+        String url = format("jdbc:presto://%s:%s", uri.getHost(), uri.getPort());
+        return DriverManager.getConnection(url, "test", null);
+    }
+
+    @Test
+    public void testCustomScheduler()
+            throws Exception
+    {
+        schedulerManager.loadScheduler();
+        Scheduler scheduler = schedulerManager.getScheduler();
+        assertEquals(scheduler.getClass(), MockScheduler.class);
+
+        scheduler.setCandidates(serverURIs);
```

Comment:
This is not testing much, we know how MockScheduler will behave and we've established on L229 that the current scheduler is a MockScheduler